apps/sms/js/thread_list_ui.js

```diff
@@ -402,10 +403,18 @@ var ThreadListUI = {
         thread.unreadCount = isRead ? 0 : 1;
         this.mark(thread.id, isRead ? 'read' : 'unread');
```

Comment:
It was part of it, initially to know which thread are to be work upon. But now, here i am storing all those thread id in threadToMark and passing that array with isRead to markReadUnread @julienw hope it clarify this